lib/user.rb

```diff
@@ -0,0 +1,43 @@
+require 'bcrypt'
+require_relative './database_connection'
+
+class User
+  def self.create(email:, password:)
+    encrypted_password = BCrypt::Password.create(password)
+
+    result = DatabaseConnection.query(
+      "INSERT INTO users (email, password) VALUES($1, $2) RETURNING id, email;", [email, password]
+    )
+    User.new(
+      id: result[0]['id'],
+      email: result[0]['email'],
+    )
+  end
+
+  def self.find(id:)
+    return nil unless id
+    result = DatabaseConnection.query(
+      "SELECT * FROM users WHERE id = $1", [id]
+    )
+    User.new(
+      id: result[0]['id'],
+      email: result[0]['email'],
+    )
+  end
+
+  def self.authenticate(email:, password:)
```

Comment:
The behaver of this method is very intresting.
I think your problem solving skills shine here! 

I would have gone about this in a different way maybe, however I don't think I fully understand the design yet.
Maybe, having it return information saying weather or not the user has an account. And then giving an option if they would like to sign up.